pandas/io/formats/style.py

```diff
@@ -233,9 +233,17 @@ def __init__(
 
     def _repr_html_(self) -> str:
         """
-        Hooks into Jupyter notebook rich display system.
+        Hooks into Jupyter notebook rich display system, which calls _repr_html_ by
+        default if an object is returned at the end of a cell.
         """
-        return self.to_html()
+        if get_option("styler.render.repr") == "html":
+            return self.to_html()
+        return None
+
+    def _repr_latex_(self) -> str:
+        if get_option("styler.render.repr") == "latex":
+            return self.to_latex()
+        return None
```

Comment:
@attack68 i think this is causing the mypy failures on CI

Reply:
will take a look